### DISCUSSION TITLE
Cosmology API

### DISCUSSION THREAD

**Discussion by marcobonici** [cloe-org/cloelib#47] [Ideas]
Hi people,
here is a point raised by @ntessore  (with a provisional answer from me)

https://github.com/cloe-library/cloelite/discussions/44#discussioncomment-11252855
https://github.com/cloe-library/cloelite/discussions/44#discussioncomment-11253264
https://github.com/cloe-library/cloelite/discussions/44#discussioncomment-11256480

What do we want to do for the Cosmology API?

Edit to link already existing issue #36

**Comment by ntessore** [comment]
As far as I can tell, there are two parts to the interaction of CLOE and the cosmology API, and I will respond to them separately.

## CLOE as a cosmology API consumer

CLOE can use the cosmoAPI to interface with other codes, e.g., at the background level. This means it won't be necessary to wrap CAMB/CLASS/CCL/jax-cosmo/etc. individually, but only have a single interface to arbitrary codes in CLOE. The wrapping of CAMB/CLASS/CCL/jax-cosmo/etc. would happen in a separate cosmoAPI-compatibility layer for each code. This has the advantage that the responsibility for maintaining the compatibility layer rests outside of CLOE, and is shared with other cosmoAPI consumers, such as GLASS.

**Comment by ntessore** [comment]
As far as I can tell, there are two parts to the interaction of CLOE and the cosmology API, and I will respond to them separately.

## CLOE as a cosmology API producer

CLOE would also be a potential provider of a cosmoAPI interface, e.g., to expose the perturbations to consumers of the cosmoAPI. This would then either require CLOE to forward its underlying background cosmology, or we split the cosmoAPI into background and perturbations.

As a concrete example, GLASS needs background quantities everywhere. It also needs the growth function for modelling IA in the NLA model.
